### PR TITLE
Update `MaterialUI` and fix imports

### PR DESCRIPTION
**Pull request by plausibly**
- The @mui version we were using seemed to be outdated, this has been fixed along with the necessary imports
- Removed some unused imports
- Removed package-lock.json since we are using yarn

If you are working on mui with your branch, to update the import just change the material UI part to be `"@mui/material/"` instead of `@material-ui/core`

### CODE REVIEW

File: client/src/components/Login.tsx [hunk 1]
```diff
@@ -1,7 +1,15 @@
 import React, { useState } from "react";
 
-import { Grid, Paper, Avatar, TextField, Button, Box } from "@material-ui/core";
-import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
+
+import Grid from "@mui/material/Grid";
```
Comment (danielduong8) [inline client/src/components/Login.tsx:4]: Not sure why you split the imports up like this, I tested it and they should still work fine if you do:

`import { ... } from "@mui/material

File: client/src/components/SignUp.tsx [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
-import CircleOutlined from "@material-ui/icons/AddCircleOutlineOutlined";
-import useMediaQuery from "@material-ui/core/useMediaQuery";
+import Grid from "@mui/material/Grid";
+import Paper from "@mui/material/Paper";
+import Avatar from "@mui/material/Avatar";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import Box from "@mui/material/Box";
+import CircleOutlined from "@mui/icons-material/AddCircleOutlineOutlined";
+import useMediaQuery from "@mui/material/useMediaQuery";
```
Comment (danielduong8) [inline client/src/components/SignUp.tsx:9]: Same comment here as above.

File: client/src/containers/AuthContainer.tsx [hunk 3]
```diff
@@ -1,13 +1,14 @@
 import React, { useState } from "react";
+ 
+import Tabs from "@mui/material/Tabs";
+import Tab from "@mui/material/Tab";
 
-import Tabs from "@material-ui/core/Tabs";
-import Tab from "@material-ui/core/Tab";
+import Box from "@mui/material/Box";
+import Paper from "@mui/material/Paper";
+import Grid from "@mui/material/Grid";
```
Comment (danielduong8) [inline client/src/containers/AuthContainer.tsx:8]: Here as well

**Review by danielduong8 — CHANGES_REQUESTED** [review]

File: client/src/components/Login.tsx (hunk 1 above)
Comment (plausibly) [reply client/src/components/Login.tsx:4]: The build time becomes insanely slow when I do that

File: client/src/components/SignUp.tsx (hunk 2 above)
Comment (plausibly) [reply client/src/components/SignUp.tsx:9]: no.

File: client/src/containers/AuthContainer.tsx (hunk 3 above)
Comment (plausibly) [reply client/src/containers/AuthContainer.tsx:8]: no!

File: client/src/components/Login.tsx (hunk 1 above)
Comment (danielduong8) [reply client/src/components/Login.tsx:4]: I looked it up and I think it has something to do with "tree-shaking".

Comment (danielduong8) [reply client/src/components/Login.tsx:4]: > What we've learned is that in order to take advantage of tree shaking, you must...
> 
> Use ES2015 module syntax (i.e. import and export).
> Ensure no compilers transform your ES2015 module syntax into CommonJS modules (this is the default behavior of the popular Babel preset @babel/preset-env - see the documentation for more details).
> Add a "sideEffects" property to your project's package.json file.
> Use the production mode configuration option to enable various optimizations including minification and tree shaking.

We could do this but it's probably just easier to separate imports.

File: client/src/App.test.tsx [hunk 4]
```diff
@@ -1,4 +1,3 @@
-import React from "react";
 import { render, screen } from "@testing-library/react";
 import App from "./App";
```
Comment (andy9775) [inline client/src/App.test.tsx:3]: Why is removing the react import necessary? If it's not, lets leave it out of this PR.

File: client/src/App.tsx [hunk 5]
```diff
@@ -1,4 +1,3 @@
-import React from "react";
 import "./App.css";
 
 import { AuthContainer } from "./containers";
```
Comment (andy9775) [inline client/src/App.tsx:3]: ditto

File: client/src/components/Login.tsx (hunk 1 above)
Comment (andy9775) [reply client/src/components/Login.tsx:4]: "Tree shaking" removes unused imports and makes your bundle size smaller (the JS file that's built) which makes for quicker asset download times and a better user experience overall.

File: client/src/containers/AuthContainer.test.tsx [hunk 6]
```diff
@@ -1,8 +1,6 @@
-import React from "react";
-
```
Comment (andy9775) [inline client/src/containers/AuthContainer.test.tsx:2]: necessary?

File: client/package.json [hunk 7]
```diff
@@ -3,8 +3,11 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@material-ui/core": "^4.12.3",
-    "@material-ui/icons": "^4.11.2",
+    "@babel/core": "^7.16.0",
+    "@emotion/react": "^11.5.0",
+    "@emotion/styled": "^11.3.0",
+    "@mui/icons-material": "^5.0.5",
```
Comment (andy9775) [inline client/package.json:9]: Can we skip the icons if we're not using them

File: client/package.json [hunk 8]
```diff
@@ -3,8 +3,11 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@material-ui/core": "^4.12.3",
-    "@material-ui/icons": "^4.11.2",
+    "@babel/core": "^7.16.0",
+    "@emotion/react": "^11.5.0",
+    "@emotion/styled": "^11.3.0",
```
Comment (andy9775) [inline client/package.json:8]: Does mui use styled components? This is a whole other library

**Review by andy9775 — CHANGES_REQUESTED** [review]
Just a few minor comments.

File: client/package.json (hunk 7 above)
Comment (plausibly) [reply client/package.json:9]: It is required and was yarn was getting mad without them

File: client/package.json (hunk 8 above)
Comment (plausibly) [reply client/package.json:8]: Yes, they give the option of installing this or `styled-components`

File: client/src/App.test.tsx [hunk 9]
```diff
@@ -1,6 +1,6 @@
...
+import React from "react";
```
Comment (andy9775) [inline client/src/App.test.tsx:3]:
```suggestion
import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
```

File: client/src/App.tsx [hunk 10]
```diff
@@ -1,6 +1,6 @@
-import React from "react";
 import "./App.css";
 
+import React from "react";
```
Comment (andy9775) [inline client/src/App.tsx:3]:
```suggestion
import React from "react";
import "./App.css";
```

File: client/src/containers/AuthContainer.test.tsx (hunk 6 above)
Comment (andy9775) [reply client/src/containers/AuthContainer.test.tsx:2]: Missed one

**Review by andy9775 — APPROVED** [review]
Just a few more comments, otherwise LGTM!

**Pull request merged**